segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -246,6 +249,80 @@ private static void validateContainerIds(Map<Integer, DebugStreamSegmentContaine
+     * @param storage                    A {@link Storage} instance where segments are stored.
+     * @param containerId                A Container Id to get the name of the metadata segment.
+     * @param backUpMetadataSegmentName  A name to the back up metadata segment.
+     * @param backUpAttributeSegmentName A name to the back attribute segment.
+     * @param executorService            A thread pool for execution.
+     * @return                           CompletableFuture which when completed will have the segments' contents copied to another segments.
+     */
+    public static CompletableFuture<Void> backUpMetadataAndAttributeSegments(Storage storage, int containerId,
+                                                                             String backUpMetadataSegmentName,
+                                                                             String backUpAttributeSegmentName,
+                                                                             ExecutorService executorService) {
+        Preconditions.checkNotNull(storage);
+        String metadataSegmentName = NameUtils.getMetadataSegmentName(containerId);
+        String attributeSegmentName = NameUtils.getAttributeSegmentName(metadataSegmentName);
+        return copySegment(storage, metadataSegmentName, backUpMetadataSegmentName, executorService)
+                        .thenAcceptAsync(x -> copySegment(storage, attributeSegmentName, backUpAttributeSegmentName,
+                                executorService));
+    }
+
+    /**
+     * Given the back up of metadata segments, this method iterates through all segments in it and gets their attributes
+     * and updates it in the container metadata segment of the given {@link DebugStreamSegmentContainer} instance.
+     * @param metadataSegments  A map of back of metadata segments along with their container Ids.
+     * @param containersMap     A map of {@link DebugStreamSegmentContainer} instances with their container Ids.
+     * @param executorService   A thread pool for execution.
+     * @throws Exception        If an exception occurred. This could be one of the following:
+     *                              * TimeoutException:     If the calls for computation(used in the method)
+     *                                                      didn't complete in time.
+     *                              * IOException     :     If a general IO exception occurred.
+     */
+    public static void updateCoreAttributes(Map<Integer, String> metadataSegments,
+                                            Map<Integer, DebugStreamSegmentContainer> containersMap,
+                                            ExecutorService executorService) throws Exception {
+        val args = IteratorArgs.builder().fetchTimeout(TIMEOUT).build();
+        SegmentToContainerMapper segToConMapper = new SegmentToContainerMapper(containersMap.size());
+
+        ArrayList<CompletableFuture<Void>> futures = new ArrayList<>();
```

Comment:
To make your life easier, I suggest you create the futures and wait for them for each container. That way if one of your containers fails to migrate, at least you know where you left off and what finished executing.

Reply:
Done.